eventuals/http.h

```diff
@@ -626,67 +852,79 @@ struct _HTTP {
 
     template <typename Arg, typename K>
     auto k(K k) && {
-      return Continuation<K>{
-          std::move(k),
-          loop_,
-          std::move(url_),
-          method_,
-          timeout_,
-          std::move(fields_)};
+      return Continuation<K>{std::move(k), loop_, std::move(request_)};
     }
 
     EventLoop& loop_;
-    std::string url_;
-    Method method_;
-    std::chrono::nanoseconds timeout_;
-    PostFields fields_;
+    Request request_;
   };
 };
 
 ////////////////////////////////////////////////////////////////////////
 
-inline auto Get(
-    EventLoop& loop,
-    const std::string& url,
-    const std::chrono::nanoseconds& timeout = std::chrono::nanoseconds(0)) {
+inline auto Client::Do(Request&& request) {
+  // TODO(benh): need 'Client::Default()'.
+  auto& loop = EventLoop::Default();
+
+  if (verify_peer_ && !request.verify_peer()) {
+    request.verify_peer_ = verify_peer_;
+  }
```

Comment:
This looks ugly! Is there a better way to override request settings with client settings?